### PR TITLE
Balance edges algos separate file

### PR DESCRIPTION
**Pull request by bsuwal**
The currently used balanced edge algorithm are in a new file, `balance_edges.jl`. 

Made a corresponding test file.

Changed the name of `weighted_kruskal_mst()` to `random_kruskal_mst()` and made the latter take the latter generate the random weights.

I introduced no new code, only renamed functions and introduced 2 new files, and moved some previously written code into them.

All of these changes are derived from #94 , but that PR is a bit too big, and has some type changes so this is the first in what I foresee about 5 small PRs that will comprise the changes in #94.

### CODE REVIEW

File: src/balance_edges.jl [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+`nodes` represent only the sub-graph on which we want to draw the MST.
+
+*Arguments:*
+- graph: Underlying Graph object
+- edges: Array of edges of the sub-graph
+- nodes: Set of nodes of the sub-graph
+
+*Returns* a BitSet of edges that form a mst.
+"""
+function random_kruskal_mst(graph::BaseGraph,
```
Comment (sunnymatt) [inline src/balance_edges.jl:71]: I like the functions being split like this! Clean and concise and easy to understand. Yay!

File: test/balance_edges.jl [hunk 2]
```diff
@@ -0,0 +1,35 @@
+using LightGraphs
+using Random
+using DataStructures
+
+""" Refer to test/graph.jl to see the test graph being loaded
+"""
+
```
Comment (sunnymatt) [inline test/balance_edges.jl:7]: Could we also write a tiny test for the `kruskal_mst` method if it's not too much trouble? 😄

**Review by sunnymatt — APPROVED** [review]
Yay for this! Just a few tiny notes (including a request for an additional test) but overall looks fantastic.

image

File: test/balance_edges.jl (hunk 2 above)
Comment (bsuwal) [reply test/balance_edges.jl:7]: of course!!! thanks for keeping me honest

Comment (bsuwal) [reply test/balance_edges.jl:4]: done and passed!

**Pull request merged**